### PR TITLE
Bar/DrinkManager safeguards, progress on User classes

### PR DESCRIPTION
**Pull request by Raeppa**
added some safeguards from exceptions and some "safer" ways to get data through methods in BarManager and Drink Manager.
ProfileForm now calls UserLoginLogic to see if the user should be logged in.
Added utility methods for User Class, made it a singleton.
Continuing work with UserLoginLogic added some hardcoded "logins" for now, now changes data in the User class when logging in/out

### CODE REVIEW

File: Comparison Engine/Comparison Engine/User Classes/User.cs [hunk 1]
```diff
@@ -17,5 +26,33 @@ public class User
...
 
+        public void ChangeUsername(string newUsername)
+        {
+            username = newUsername;
+            SaveUserInfo();
+        }
+
+        public void ChangeFaveBar(Bar newBar)
+        {
+            favoriteBar = newBar;
```
Comment (AdamBeresnev) [inline Comparison Engine/Comparison Engine/User Classes/User.cs:37]: I think if we want to add a feature to have a favorite bar, then maybe having more than one would make more sense

**Review by AdamBeresnev — APPROVED** [review]
Don't see any problems, except in my opinion the current User.cs should be renamed to UserManager.cs and User.cs should be reserved for storing different users(although I might be wrong, depending on how we decide to store user information)

File: Comparison Engine/Comparison Engine/User Classes/User.cs (hunk 1 above)
Comment (Raeppa) [reply Comparison Engine/Comparison Engine/User Classes/User.cs:37]: That's definitely fair, will fix that

**Comment by Raeppa** [comment]
I agree that it could be renamed to UserManager.cs, however, it shouldn't be reserved to store different users, as I don't think a user should be able to access other user's information at any point. And any info from other users that we might need in the future, will probably be stored not in the code somewhere, but in some sort of database

**Comment by AdamBeresnev** [comment]
> I agree that it could be renamed to UserManager.cs, however, it shouldn't be reserved to store different users, as I don't think a user should be able to access other user's information at any point. And any info from other users that we might need in the future, will probably be stored not in the code somewhere, but in some sort of database

What I meant was that User.cs would be a class like Bar/Drink and users would be objects of that class

**Pull request merged**